pkg/beyla/os.go

```diff
@@ -21,3 +25,85 @@ func CheckOSSupport() error {
 	}
 	return nil
 }
+
+type osCapabilitiesError uint64
+
+func (e *osCapabilitiesError) Set(c helpers.OSCapability) {
+	*e |= 1 << c
+}
+
+func (e *osCapabilitiesError) Clear(c helpers.OSCapability) {
+	*e &= ^(1 << c)
+}
+
+func (e osCapabilitiesError) IsSet(c helpers.OSCapability) bool {
+	return e&(1<<c) > 0
+}
+
+func (e osCapabilitiesError) Empty() bool {
+	return e == 0
+}
+
+func (e osCapabilitiesError) Error() string {
+	if e == 0 {
+		return ""
+	}
+
+	var sb strings.Builder
+
+	sep := ""
+
+	for i := helpers.OSCapability(0); i <= unix.CAP_LAST_CAP; i++ {
+		if e.IsSet(i) {
+			sb.WriteString(sep)
+			sb.WriteString(i.String())
+
+			sep = ", "
+		}
+	}
+
+	return fmt.Sprintf("the following capabilities are required: %s", sb.String())
```

Comment:
Super-nitpick: after `var sb strings.Builder` definition, just add:

```go
sb.WriteString("the following capabilities are required: ")
```

Then you can just return:
```go
return sb.String()
```